source/common/network/matching/inputs.h

```diff
@@ -252,6 +255,86 @@ class ApplicationProtocolInputFactory
   ApplicationProtocolInputFactory() : BaseFactory("application_protocol") {}
 };
 
+template <class MatchingDataType>
+class AuthenticatedInput : public Matcher::DataInput<MatchingDataType> {
+public:
+  explicit AuthenticatedInput(
+      envoy::extensions::matching::common_inputs::network::v3::AuthenticatedInput::Field field)
+      : field_(field) {}
+
+  Matcher::DataInputGetResult get(const MatchingDataType& data) const override {
+    const auto& ssl = data.ssl();
+    if (!ssl) {
+      return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable, absl::nullopt};
+    }
+
+    switch (field_) {
+      PANIC_ON_PROTO_ENUM_SENTINEL_VALUES;
+    case envoy::extensions::matching::common_inputs::network::v3::AuthenticatedInput::URI_SAN: {
+      const auto& uri = ssl->uriSanPeerCertificate();
+      if (!uri.empty()) {
+        return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable,
+                absl::StrJoin(uri, ",")};
+      }
+      return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable, absl::nullopt};
+    }
+    case envoy::extensions::matching::common_inputs::network::v3::AuthenticatedInput::DNS_SAN: {
+      const auto& dns = ssl->dnsSansPeerCertificate();
+      if (!dns.empty()) {
+        return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable,
+                absl::StrJoin(dns, ",")};
+      }
+      return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable, absl::nullopt};
+    }
+    case envoy::extensions::matching::common_inputs::network::v3::AuthenticatedInput::SUBJECT: {
+      const auto& subject = ssl->subjectPeerCertificate();
+      if (!subject.empty()) {
+        return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable,
+                ssl->subjectPeerCertificate()};
+      }
+      return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable, absl::nullopt};
+    }
+    case envoy::extensions::matching::common_inputs::network::v3::AuthenticatedInput::
+        FIELD_TYPE_UNSPECIFIED: {
+      PANIC("unhandled value");
```

Comment:
If we are going with this enum approach I would think we could do this switching on construction time and not on each match attempt? That would avoid having this panic branch in here